### PR TITLE
auto save on page navigation

### PR DESCRIPTION
**Pull request by prabhjyotsingh**
This should take care of accidental browser close actions.

### CODE REVIEW

**Comment by corneadoug** [comment]
This is working well.

While testing this branch I found those, and I think it could be taken care of here:
* When we fast click on notebook loading to home or fast refresh on notebook loading, we get a la few undefined errors in the console (values need to be checked before being used)
* When the timeout of startSaveTimer run after we changed page, we get an undefined error too.

**Comment by prabhjyotsingh** [comment]
@corneadoug  thanks for the quick feedback.
This should take care of those error.

**Comment by corneadoug** [comment]
I was more thinking about handling it in the saveNote function directly

**Comment by prabhjyotsingh** [comment]
have revert old changes, and made fix on $scope.saveNote

File: zeppelin-web/src/app/notebook/notebook.controller.js [hunk 1]
```diff
@@ -150,7 +152,7 @@ angular.module('zeppelinWebApp').controller('NotebookCtrl', function($scope, $ro
   };
 
   $scope.killSaveTimer = function() {
```
Comment (corneadoug) [inline zeppelin-web/src/app/notebook/notebook.controller.js]: This should be changed to var killsaveTimer = function()
There is no point of having it in the $scope.

Comment (prabhjyotsingh) [inline zeppelin-web/src/app/notebook/notebook.controller.js]: I've seen a lot of other function/variables (like renderAngular in paragraph.controller.js) that should be removed from $scope, and plan to do that in a separate clean up PR.

Comment (corneadoug) [inline zeppelin-web/src/app/notebook/notebook.controller.js]: Yeah, there is a lot of code that need refactoring.
But there is no point in adding more work to that future task.

Comment (prabhjyotsingh) [inline zeppelin-web/src/app/notebook/notebook.controller.js]: Sure thing.

**Comment by Leemoonsoo** [comment]
@prabhjyotsingh @corneadoug Is it ready to merge?
Can you solve conflicts with master branch?

**Comment by prabhjyotsingh** [comment]
@Leemoonsoo have resolved conflicts. Thanks for the review.

**Comment by prabhjyotsingh** [comment]
@corneadoug @Leemoonsoo does this fix looks good ?

**Comment by prabhjyotsingh** [comment]
@corneadoug @Leemoonsoo do i need to add something as well, or is there a case that has been missed out ?

**Comment by corneadoug** [comment]
@prabhjyotsingh Sorry, kinda forgot about this PR being ready.
I see that one test was squashed also what it was testing still exist, could you fix that? ``it('should first call killSaveTimer() when calling startSaveTimer()'``

**Comment by prabhjyotsingh** [comment]
@corneadoug  I thought to fix that with ZEPPELIN-294 (/pull/292). Have made the relevant changes here for the review.

**Comment by prabhjyotsingh** [comment]
I see all builds "SUCCESS". So, does this mean this is a false positive error ?

https://s3.amazonaws.com/archive.travis-ci.org/jobs/82533319/log.txt
screen shot 2015-09-29 at 10 41 56 am

**Comment by corneadoug** [comment]
You can try to close/reopen the PR to relaunch the CI

**Comment by prabhjyotsingh** [comment]
closing this, will resolve travis build issue, and then re-open

**Pull request closed without merging**